contrib/scalapblib/src/ScalaPBModule.scala

```diff
@@ -94,35 +93,27 @@ trait ScalaPBModule extends ScalaModule {
   def scalaPBUnpackProto: T[PathRef] = T {
     val cp = scalaPBProtoClasspath()
     val dest = T.dest
-    cp.foreach { ref =>
-      val baseUri = "jar:" + ref.path.toIO.getCanonicalFile.toURI.toASCIIString
-      val jarFs =
-        FileSystems.newFileSystem(URI.create(baseUri), new java.util.HashMap[String, String]())
-      try {
-        import scala.collection.JavaConverters._
-        jarFs.getRootDirectories.asScala.foreach { r =>
-          Files.walkFileTree(
-            r,
-            new SimpleFileVisitor[Path] {
-              override def visitFile(f: Path, a: BasicFileAttributes) = {
-                if (f.getFileName.toString.endsWith(".proto")) {
-                  val protoDest = dest.toNIO.resolve(r.relativize(f).toString)
-                  Files.createDirectories(protoDest.getParent)
-                  Files.copy(
-                    f,
-                    protoDest,
-                    StandardCopyOption.COPY_ATTRIBUTES,
-                    StandardCopyOption.REPLACE_EXISTING
-                  )
-                }
-                super.visitFile(f, a)
+    val buffer = new Array[Byte](8192)
+    cp.iterator.foreach { ref =>
+      Using(new ZipInputStream(ref.path.getInputStream)) { zip =>
+        var entry = Option(zip.getNextEntry)
+        while (entry.nonEmpty) {
+          val name = entry.get.getName
+          if (name.endsWith(".proto")) {
+            val protoDest = os.Path(os.RelPath(name), dest)
```

Comment:
I'd suggest `os.Subpath` which ensures we won't step outside the parent dir.

```suggestion
            val protoDest = dest / os.SubPath(name)
```